collector/hot_threads.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package collector
+
+import (
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	defaultHotThreadsLabels = []string{"node", "thread_name", "thread_id"}
+
+	defaultHotThreadsLabelValues = func(HotThreads string) []string {
+		return []string{
+			HotThreads,
+		}
+	}
+	NODE_OUTPUT_SEPERATOR = ":::"
```

Comment:
I'm not sure why all of this data is being parsed by regex. It seems error prone. I think some tests would be very useful.